src/sentry/api/invite_helper.py

```diff
@@ -170,19 +183,36 @@ def accept_invite(self, user=None):
         if self.member_already_exists:
             self.handle_member_already_exists()
             om.delete()
-        else:
-            om.set_user(user)
-            om.save()
-
-            create_audit_entry(
-                self.request,
-                actor=user,
-                organization=om.organization,
-                target_object=om.id,
-                target_user=user,
-                event=AuditLogEntryEvent.MEMBER_ACCEPT,
-                data=om.get_audit_log_data(),
-            )
+            return
+
+        try:
+            provider = AuthProvider.objects.get(organization=om.organization)
+        except AuthProvider.DoesNotExist:
+            provider = None
+
+        # If SSO is required, check for valid AuthIdentity
+        if provider and not provider.flags.allow_unlinked:
+            try:
+                authId = AuthIdentity.objects.get(user=user)
```

Comment:
You should filter by `provider` here as well. Not sure whether it's possible to have > 1 `AuthIdentity`, but `.get` will fail if > 1 rows are returned.

Reply:
Good catch. I've updated it.